src/main/java/io/cdap/plugin/successfactors/source/service/SuccessFactorsService.java

```diff
@@ -131,4 +159,213 @@ private InputStream callEntityMetadata() throws TransportException {
+   *
+   * @param edm  SuccessFactors service entity metadata
+   * @param skip number of rows to skip
+   * @param top  number of rows to fetch
+   * @return {@code ODataFeed}
+   * @throws TransportException             any http client exceptions are wrapped under it.
+   * @throws SuccessFactorsServiceException any OData service based exception is wrapped under it.
+   */
+  public ODataFeed readServiceEntityData(Edm edm, Long skip, Long top)
+    throws SuccessFactorsServiceException, TransportException {
+
+    SuccessFactorsEntityProvider serviceHelper = new SuccessFactorsEntityProvider(edm);
+    try (InputStream dataStream = callEntityData(skip, top)) {
+
+      EdmEntitySet entity = serviceHelper.getEntitySet(pluginConfig.getEntityName());
+      // compile raw data to ODataFeed type
+      ODataFeed dataFeed;
+      if (pluginConfig.getExpandOption() != null) {
+        dataFeed = EntityProvider
+          .readFeed(MediaType.APPLICATION_JSON, entity, filterExpandedEntityData(dataStream),
+                    EntityProviderReadProperties.init().build());
+      } else {
+        dataFeed = EntityProvider
+          .readFeed(MediaType.APPLICATION_JSON, entity, dataStream, EntityProviderReadProperties
+            .init().build());
+      }
+
+      if (dataFeed != null) {
+        if (pluginConfig.getPaginationType().equals(SERVER_SIDE)) {
+          String nextLink = dataFeed.getFeedMetadata().getNextLink();
+          if (nextLink != null) {
+            nextUrl = nextLink;
+            LOG.info("Next page url: {}", nextLink);
+          }
+        }
+        return dataFeed;
+      }
+
+    } catch (EdmException | EntityProviderException | IOException ex) {
+      String errMsg = ResourceConstants.ERR_RECORD_PROCESSING.getMsgForKeyWithCode(pluginConfig.getEntityName());
+      throw new SuccessFactorsServiceException(errMsg, ex);
+    } catch (TransportException te) {
+      String errMsg = ResourceConstants.ERR_RECORD_PULL.getMsgForKeyWithCode(pluginConfig.getEntityName());
+      errMsg += ExceptionParser.buildTransportError(te);
+      throw new TransportException(errMsg, te);
+    } catch (SuccessFactorsServiceException ose) {
+      String errMsg = ResourceConstants.ERR_RECORD_PULL.getMsgForKeyWithCode(pluginConfig.getEntityName());
+      errMsg += ExceptionParser.buildSuccessFactorsServiceError(ose);
+      throw new SuccessFactorsServiceException(errMsg, ose);
+    }
+    return null;
+  }
+
+  /**
+   * Calls the SAP SuccessFactors service entity to fetch the data from the given range
+   *
+   * @param skip number to rows to skip
+   * @param top  number to rows to fetch
+   * @return {@code InputStream}
+   * @throws TransportException             any http client exceptions are wrapped under it.
+   * @throws SuccessFactorsServiceException any OData service based exception is wrapped under it.
+   */
+  private InputStream callEntityData(Long skip, Long top)
```

Comment:
Annotate skip, top params as nullable

Reply:
Added `@Nullable` annotation to the skip and top params in `callEntityData()` b281d64bd8cde098f1e60f7ab44c12178af0e71d